resource/reapi/bindings/go/src/fluxcli/reapi_cli.go

```diff
@@ -14,226 +14,260 @@ package fluxcli
 #include "reapi_cli.h"
 */
 import "C"
+import (
+	"fmt"
+	"unsafe"
+)
 
 type (
 	ReapiCtx C.struct_reapi_cli_ctx_t
 )
 
-/*! Create and initialize reapi_cli context
-reapi_cli_ctx_t *reapi_cli_new ();
-*/
-
+// NewReapiCli creates a new resource API client
+// reapi_cli_ctx_t *reapi_cli_new ();
 func NewReapiCli() *ReapiCtx {
 	return (*ReapiCtx)(C.reapi_cli_new())
 }
```

Comment:
In your review https://github.com/flux-framework/flux-sched/pull/1008#discussion_r1193018843 you made a very good suggestion:

> In practice, this may mean just having a go struct as a wrapper around this context that is used, and then removing the need to include it as the first parameter of every function.

I don't see a go wrapper struct implemented. Could you add that?

Reply:
Yes! I forget why I didn't do it - I think I was worried about adding too much complexity to the review. But I would prefer this design as well.